src/prompt/files_prompt.rs

```diff
@@ -134,16 +139,12 @@ impl<'a> FilesPrompt<'a> {
                     };
                 }
                 Some((KeyCode::Down, _, _, true)) => {
-                    let total = self.options.len() as u16 + 1;
-
-                    self.focused_index += total.saturating_sub(1);
+                    self.focused_index += (take as u16 + 1).saturating_sub(1);
```

Comment:
`(take as u16 + 1).saturating_sub(1)` is the same as `take as u16` here.

Reply:
Oh, duh, apologies.